Allow unified search filtering and bump @nextcloud/vue

skjnldsv opened the pull request:
![Capture d’écran_2020-08-06_08-34-04](https://user-images.githubusercontent.com/14975046/89499096-b98d6c00-d7bf-11ea-871e-d2f12474f7fa.png)

skjnldsv committed:
Allow unified search filtering

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

skjnldsv commented on core/src/components/UnifiedSearch/SearchFilter.vue:
```diff
@@ -0,0 +1,77 @@
+	margin-right: 5px;
+	padding: 3px 8px;
+	border-radius: 1em;
+	background-color: var(--color-background-darker);
+
+	&:active,
+	&:focus,
+	&:hover {
+		background-color: var(--color-background-hover);
+	}
```
Here you go @jancborchardt, choose whatever you want :D

georgehrke commented:
OT: Are there plans to allow search providers to support their own filters?
For events i could think of: `before: 2020`, `at: location123`

skjnldsv commented:
> OT: Are there plans to allow search providers to support their own filters?
> For events i could think of: `before: 2020`, `at: location123`

You can do it already, i'm only filtering in:xxxx searches, anything else will be pass through the search.

skjnldsv committed:
Fix search placeholder animation & dark theme compatibility

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

jancborchardt reviewed (dismissed):
As discussed, the filters are now a bit too present – more present than the results themselves.

Since it’s basically a guide for very advanced features, they can go into a 3-dot menu on the right inside the search field, which opens a menu with these filters. That way they are not in the way, people can still easily reach them via mouse, and they are also learnable

jancborchardt commented:
Looks much better with the menu! :) Only: If the icon is the same for all, we don’t need the icon. Or we show the relevant app icon there. :)

skjnldsv commented:
> Or we show the relevant app icon there

Would be nice, but we don't provide a unified way to get the apps icons :)
Something for 21? :grin:

nickvergessen commented:
But search filters are not bound to apps... So therw is no 1-to-1 mapping to app icons

skjnldsv commented:
> But search filters are not bound to apps... So therw is no 1-to-1 mapping to app icons

Exactly :)
We'd need a new entry in the api :wink:

skjnldsv commented:
Let get this in, the popup fix will come from the vue library. It's irrelevant to this pr! :rocket:

nickvergessen commented:
> Conflicting files
core/js/dist/install.js
core/js/dist/login.js
core/js/dist/login.js.map
core/js/dist/main.js
core/js/dist/main.js.map
core/js/dist/maintenance.js
core/js/dist/maintenance.js.map
core/js/dist/recommendedapps.js
core/js/dist/recommendedapps.js.map
core/js/dist/unified-search.js
core/js/dist/unified-search.js.map

nickvergessen commented:
/compile amend /

skjnldsv commented:
> /compile amend /

This is not a rebase bot!
Please wait for my rebase

nickvergessen commented:
Right, i thought compile could fix it... Too long vacation 😅
Just wanted to make sure we can actually merge it tomorrow instead of having to rebase in daily traffic

skjnldsv commented:
> Just wanted to make sure we can actually merge it tomorrow instead of having to rebase in daily traffic

I'm about to push alongisde a new vue release :)

skjnldsv commented:
> I'm about to push alongisde a new vue release :)

DOne :)

skjnldsv commented:
/compile amend /

skjnldsv commented:
Needs a new nv/vue release as sidebars are buggued, see https://github.com/nextcloud/nextcloud-vue/pull/1303/

skjnldsv committed:
Bump @nextcloud/vue to v2.6.3

Signed-off-by: John Molakvoæ (skjnldsv) <skjnldsv@protonmail.com>

georgehrke reviewed (approved):
👍 Tested and works

skjnldsv commented:
Let's get this in!

georgehrke reviewed (approved):
Still working as expected 👍

georgehrke commented:
Only minor issue is the text overflow here, but let's fix that in a follow up PR:
![3EC70324-DC30-46EE-8656-A944D96965D0](https://user-images.githubusercontent.com/1250540/91823335-587b7b80-ec39-11ea-820c-6c3472b1ae84.png)

skjnldsv commented:
> Only minor issue is the text overflow here, but let's fix that in a follow up PR:

Not sure this is a fix for server. More like a vue component fix.

georgehrke commented:
Opened an issue there: https://github.com/nextcloud/nextcloud-vue/issues/1367

skjnldsv commented:
Needs https://github.com/nextcloud/nextcloud-vue/pull/1368

danxuliu committed:
Adjust acceptance tests to changes in @nextcloud/vue 2.6.3

The menu button and the menu of a share are no longer direct childs of
the actions of the share row. The menu button is now a child of a
".trigger" element, while the menu is a direct child of the body and has
an id defined in the "aria-describedby" attribute of the ".trigger"
element.

In XPath 1.0 it does not seem possible to "backreference" a value or
create variables, so when the share menu or one of its item is needed
now the ".trigger" element is first found and then its XPath expression
is used to compose its "aria-describedby" attribute in the XPath
expression for the menu.

Signed-off-by: Daniel Calviño Sánchez <danxuliu@gmail.com>

nickvergessen commented:
> ### Conflicting files
> core/src/views/UnifiedSearch.vue

:see_no_evil:

nickvergessen reviewed (approved):
:shark:
